cloud-function/src/build-sitemap.ts

```diff
@@ -0,0 +1,39 @@
+import { readFile, stat, writeFile } from "node:fs/promises";
```

Comment:
> Generally, why not create the json instead of the .txt?

I went for text format, because [text sitemaps](https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap#text) are a thing and it's a simpler format with fewer assumptions than a JSON, although I'm not technically adhering to the sitemap convention of using fully qualified URLs.

> I mainly want be aware of what's the contract between building and the cloud function and I think the json makes a better api.

Are you suggesting to publish a `sitemap.json` with a mapping of normalized path to canonical path, to be used by other consumers potentially? That would actually be my concern that other parties start using this JSON if they come across it. On the other hand, would we change it?

> If there's a strong reason to keep this I'd just have ask to use a Object instead of the Map since we're putting it into and Object any way.

Done in d2e58ca1d4.

Reply:
Yes I was suggesting to build the same `sitemap.json` as in this file. I wouldn't call it sitemap though.

I would not _publish_ that file. Is there a benefit for having the sitemap as text and as XML? If we go for only txt sitemap I'm all in for this. Otherwise this ads complexity and we potentially have to support two sitemaps formats from now on.

Reply:
I don't mind renaming, and I could surely build that map as part of the build, but I don't think it reduces complexity:

1. I need to get all URLs from all sitemaps, so either I gather them as a side-effect of building the sitemaps, or I go over all (gzip-compressed) XML sitemaps later, parsing out the URLs.
2. We build and deploy the Cloud Function after we sync the build to Cloud Storage, and `gsutil` doesn't support exclusion very well (cf. the `plain.html` files that we sync), so we might need to prevent access to that file in the Cloud Function.
3. We definitely want to keep using the XML sitemap for search engines (as advised in `robots.txt`), because the text format doesn't support modification dates.
4. We would need to extract the `normalizePath()` function into a lib shared by build and cloud function, and I assume this would no longer be possible with rari. We could then duplicate the logic, but it's still risky.

Overall, I would recommend to keep the current approach.